happi/cli.py

```diff
@@ -97,223 +47,314 @@ def happi_cli(args):
+    else:
+        clone_source = None
+        # Keep Device at registry for backwards compatibility but filter
+        # it out of new devices options
+        options = os.linesep.join(
+            [k for k, _ in registry.items() if k != "Device"]
+        )
+        ctnr_prompt = (
+            'Please select a container, or press enter for generic '
+            f'Ophyd Device container: {os.linesep}{options}'
+            '{}{}Selection'.format(os.linesep, os.linesep)
+        )
+        logger.debug(ctnr_prompt)
+        response = click.prompt(ctnr_prompt, default='OphydItem')
+        if response and response not in registry:
+            logger.info(f'Invalid device container {response}')
+            return
+
+    container = registry[response]
+    logger.debug(f'Contaner selected: {container.__name__}')
+
+    # Collect values for each field
+    kwargs = {}
+    for info in container.entry_info:
+        item_value = prompt_for_entry(info, clone_source=clone_source)
+        click.echo(f'Selecting value: {item_value}')
+
+        try:
+            info.enforce_value(item_value)
+            kwargs[info.key] = item_value
+        except Exception:
+            logger.info(f'Invalid value {item_value}')
+
+    device = client.create_device(container, **kwargs)
+    device.show_info()
+    if click.confirm('Please confirm the device info is correct'):
+        logger.info('Adding device')
         device.save()
-        device.show_info()
-    elif args.cmd == 'load':
-        logger.debug('Starting load block')
-        logger.info(f'Creating shell with devices {args.device_names}')
-        devices = {}
-        names = " ".join(args.device_names)
-        names = names.split()
-        for name in names:
-            devices[name] = client.load_device(name=name)
-
-        from IPython import start_ipython  # noqa
-        start_ipython(argv=['--quick'], user_ns=devices)
-    elif args.cmd == "update":
-        # parse input
-        input_ = " ".join(args.json).strip()
-        if input_ == "-":
-            items_input = json.load(sys.stdin)
+    else:
+        logger.info('Aborting')
+
+
+@happi_cli.command()
+@click.argument('name')
+@click.argument('edits', nargs=-1, type=str)
+@click.pass_context
+def edit(ctx, name, edits):
+    """
+    Change an existing entry by applying EDITS of the form: field=value
+    to the item of name NAME.
+    """
+    # retrieve client
+    client = ctx.obj
+
+    logger.debug('Starting edit block')
+    device = client.find_device(name=name)
+    for edit in edits:
+        field, value = edit.split('=', 1)
+        # validate field
+        try:
+            field = is_valid_identifier_not_keyword(field)
+        except ValueError:
+            logger.error(f'field ({field}) is not a valid keyword')
+            raise click.Abort()
+
+        # validate field depending on enforce type
+        if device._info_attrs[field].enforce in (dict, list):
```

Comment:
Now this is a long-standing bugfix!

I think you could also do `happi edit item 'dict={"...": "..."}'` to avoid having the shell get in the way of quote interpretation (note the single quotes surrounding)

Reply:
That's right, dealing with quotes is something I was mulling over for a while.  I'm not sure which way users would like more (extra quotes or escaping quotes), but I guess that'll shake out from here.